specs-go/config.go

```diff
@@ -22,11 +22,24 @@ type Spec struct {
 	Hooks Hooks `json:"hooks"`
 	// Annotations is an unstructured key value map that may be set by external tools to store and retrieve arbitrary metadata.
 	Annotations map[string]string `json:"annotations,omitempty"`
-
+	// KernelConfig specifies kernel-related configuration for virtual-machine-based runtimes.
+	KernelConfig KernelConfig `json:"kernelConfig"`
+	// ImagePath is the absolute path to the root filesystem image on the host which can be used by a virtual machine-based runtime.
+	ImagePath string `json:"imagePath"`
 	// Linux is platform specific configuration for Linux based containers.
 	Linux Linux `json:"linux" platform:"linux"`
```

Comment:
On Wed, May 18, 2016 at 10:40:15AM -0700, Michael Crosby wrote:
> >  	// Linux is platform specific configuration for Linux based containers.
> >  	Linux Linux `json:"linux" platform:"linux"`
>
> not sure we need that type of granularity in the fields when we have
> the platform struct in the spec

I'm not sure about the granularity.  If the new VM fields are intended
to support VMs running different guest platforms (e.g. guest Linux,
guest Windows, …) then generic ‘virtual-machine’ namespacing sounds
good.  If the new fields are only for Linux guests, then specific
‘linux-virtual-machine’ sounds good.

In the latter case, changing the existing ‘linux’ to ‘linux-container’
makes sense (no need to make the container implementation special),
but maybe isn't worth breaking backwards compatibility.  It does mean
that *new* entries should be as specific as possible to avoid future
ambiguity sticking around for backwards compatibility.

And with VM's in the picture, we may have to overhaul the ‘platform’
entry to allow for a host/guest distinction.